src/SFML/Graphics/ImageLoader.cpp

```diff
@@ -227,35 +227,34 @@ bool ImageLoader::saveImageToFile(const std::string& filename, const std::vector
     if (!pixels.empty() && (size.x > 0) && (size.y > 0))
     {
         // Deduce the image type from its extension
-        if (filename.size() > 3)
+
+        // Extract the extension
+        const std::size_t dot = filename.find_last_of('.');
+        const std::string extension = dot != std::string::npos ? toLower(filename.substr(dot + 1)) : "";
```

Comment:
I think the check for a valid `dot` index should be done first in an `if` embracing all the following code. What do you think ?
```c++
if (dot != std::string::npos)
{
    const std::string extension = toLower(filename.substr(dot + 1));
    if (extension == "bmp")
    //...
}
```